src/postgres/commands/PostgresAccountWizard/PostgresServerFirewallStep.ts

```diff
@@ -0,0 +1,30 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as publicIp from 'public-ip';
+import { QuickPickItem } from 'vscode';
+import { AzureWizardPromptStep } from 'vscode-azureextensionui';
+import { ext } from '../../../extensionVariables';
+import { localize } from '../../../utils/localize';
+import { IPostgresWizardContext } from './IPostgresWizardContext';
+
+export class PostgresServerFirewallStep extends AzureWizardPromptStep<IPostgresWizardContext> {
+
+    public async prompt(wizardContext: IPostgresWizardContext): Promise<void> {
+
+        const ip: string = await publicIp.v4();
```

Comment:
If you await this here and it takes a while, the user won't see anything while its executing. Instead, we want to pass the _promise_ to `showQuickPick` and VS Code will show a little loading animation while the promise executes. Here's how I usually do it:

Define a separate method to get the picks:
```
private async getPicks(): Promise<IAzureQuickPickItem<boolean>[]> {
    const ip: string = await publicIp.v4();
    return [
        { label: localize('addFirewallRule', 'Add firewall rule for IP "{0}"', ip), data: true },
        { label: localize('skipFireWallRule', '$(clock) Skip for now'), data: false }
    ];
}
```

When calling `showQuickPick`, call `getPicks` _without_ `await`
```
wizardContext.addFirewall = (await ext.ui.showQuickPick(this.getPicks(), { placeHolder })).data;
```
NOTE: I also refactored to use `data` instead of checking `=== yes` since we don't know the value of `yes` until after the promise finishes